VERSIONING.md

```diff
@@ -33,3 +33,35 @@ Compose version | Kotlin version
 1.0.0 | 1.5.31
 1.1.1 | 1.6.20
 1.2.0 | 1.7.10 - 1.7.20 (Desktop, Android), 1.7.10 (JS)
+
+### Relation to Jetpack Compose and Compose Multiplatform release cycle
+
+Compose Multiplatform shares a lot of code with [Jetpack Compose](https://developer.android.com/jetpack/compose) for Android developed by Google.
+We keep our release cycles aligned, so the common part is properly tested and stabilized.
+
+When the new minor version of Jetpack Compose is released, we pick the release commit, use it as a base for [Compose Multiplatform fork](https://github.com/JetBrains/androidx), finish new platform features, stabilize all platforms, and release Compose Multiplatform after some time.
+The gap between the Compose Multiplatform release and the Jetpack Compose release is usually 1-3 months.
+
+When you build your application for Android, the artifacts published by Google are used.
+
+In the next table you can see versions of the corresponding Jetpack Compose libraries.
+These are the versions on which we based Compose Multiplatform, and the versions used when you build your application for Android.
+
+Compose Multiplatform | Jetpack Compose Compiler | Jetpack Compose Runtime, UI, Foundation, Material | Jetpack Compose Material 3
+--- |--------------------|---------------------------------------------------| ---
+1.0.0 | 1.1.0-rc03 | 1.1.0-rc03                                        | 1.0.0-alpha03
+1.1.1 | 1.1.0 | 1.1.0                                             | 1.0.0-alpha05
+1.2.0 | 1.3.2 | 1.2.1                                             | 1.0.0-alpha14
+
+You may want to use [a cutting edge Compose Compiler](https://developer.android.com/jetpack/androidx/releases/compose-kotlin) published by Google for your Compose Multiplatform project.
+For example, when the new Kotlin is released,  and there is the Jetpack Compose Compiler that supports it,
+and Compose Multiplatform still doesn't support it.
+
+In that case you can set `kotlinCompilerPlugin` in the `compose` section of your `build.gradle.kts` file:
+
+```kotlin
+compose {
+    kotlinCompilerPlugin.set("androidx.compose.compiler:compiler:1.3.1")
```

Comment:
If we release the new version of plugin every time when we release a new version of Compose Compiler, then (considering that we also support multiple version of Kotlin now) this information won't be relevant to majority of users. There won't a case when they needed to overwrite it separately. Upgrading to a newer patch of Compose Gradle Plugin (1.2.0 -> 1.2.1) is similar to upgrading Compose Compiler.

Also, if we describe it, we should also describe differences in our Compose Compiler and Jetpack Compose Compiler, and make a separate list of all our versions of Compiler. The Jetpack Compose users will think "by default" that they should pick  a version from [here](https://developer.android.com/jetpack/androidx/releases/compose-kotlin), which is not true.

This feature I believe is useful in these cases:
- asking users to use a different compiler to investigate an issue
- we released a new compiler, but still haven't released a new plugin (we have an error on CI, for example)
- users don't want to upgrade to major version of Compose (we released 1.3.0 with Kotlin 1.7.30 support, but didn't release 1.2.3)
- we realize that our current approach has flaws, or users actively use it instead of our current approach

So, maybe we will see how it works in battle, suggesting this "hidden" feature in issues, discussions and Slack before we describe it as another way to support new Kotlin? Having two ways (one - upgrading Compose Plugin, two - overriding a separate field) complicates things, and we have to be sure that another way adds enough value. 

On the other hand, using third-party (Jetpack) compiler is more useful feature, because Jetpack Compose Compiler is released more frequently and earlier.